test/sequential/test-runner-run-inspect.js

```diff
@@ -0,0 +1,240 @@
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'bignumber',
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'negativenumber',
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'bignumberfunc'
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'strfunc',
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 0,
+    expectedInitialPort: 0,
+  });
+
+  await defaultPortCase;
+
+  port = debuggerPort + offset++ * 5;
+  await spawnRunner({
+    execArgv: ['--inspect'],
+    inspectPort: port + 2,
+    expectedInitialPort: port + 2,
+  });
+}
+
+const badPortError = { name: 'RangeError', code: 'ERR_SOCKET_BAD_PORT' };
+function runTest() {
+  const { run } = require('node:test');
+  let inspectPort = 'inspectPort' in process.env ? Number(process.env.inspectPort) : undefined;
+  let expectedError;
+
+  if (process.env.inspectPort === 'addTwo') {
+    inspectPort = common.mustCall(() => { return process.debugPort += 2; });
+  } else if (process.env.inspectPort === 'string') {
+    inspectPort = 'string';
+    expectedError = badPortError;
+  } else if (process.env.inspectPort === 'null') {
+    inspectPort = null;
+  } else if (process.env.inspectPort === 'bignumber') {
+    inspectPort = 1293812;
+    expectedError = badPortError;
+  } else if (process.env.inspectPort === 'negativenumber') {
+    inspectPort = -9776;
+    expectedError = badPortError;
+  } else if (process.env.inspectPort === 'bignumberfunc') {
+    inspectPort = common.mustCall(() => 123121);
+    expectedError = badPortError;
+  } else if (process.env.inspectPort === 'strfunc') {
+    inspectPort = common.mustCall(() => 'invalidPort');
+    expectedError = badPortError;
+  }
+
+  const stream = run({ files: [__filename], inspectPort });
+  if (expectedError) {
+    stream.once('test:fail', common.mustCall(({ error }) => {
+      assert.deepStrictEqual({ name: error.cause.name, code: error.cause.code }, expectedError);
+    }));
+  } else {
+    stream.once('test:fail', ({ error }) => { throw error; });
+  }
+}
+
+
+function assertParams() {
+  const { expectedPort, expectedInitialPort, expectedHost } = process.env;
+  const debugOptions =
+    require('internal/options').getOptionValue('--inspect-port');
+
+  if ('expectedPort' in process.env) {
+    assert.strictEqual(process.debugPort, +expectedPort);
+  }
+
+  if ('expectedInitialPort' in process.env) {
+    assert.strictEqual(debugOptions.port, +expectedInitialPort);
+  }
+
+  if ('expectedHost' in process.env) {
+    assert.strictEqual(debugOptions.host, expectedHost);
+  }
+  process.exit();
+}
+
+if (!process.env.testProcess && !isTopLevel) {
+  testRunnerMain().then(common.mustCall());
```

Comment:
nit: you could rename the file to `.mjs`, and use TLA instead of creating an async `testRunnerMain`.